src/restrict.jl

```diff
@@ -0,0 +1,35 @@
+# Originally from Images.jl
+restrict(img::AxisArray, ::Tuple{}) = img
+
+function restrict(img::AxisArray, region::Dims)
+    inregiont = ntuple(i->in(i, region), ndims(img))
```

Comment:
Originally this is:

```diff
-     inregiont = ntuple(i->in(i, region), ndims(img))
+     inregion = falses(ndims(img))
+     inregion[[region...]] .= true
+     inregiont = (inregion...,)::NTuple{ndims(img),Bool}
```

It's more efficient to use `ntuple` here:


```julia
imgcol = rand(RGB{Float64}, 5, 6)
imgcolax = AxisArray(imgcol, :y, :x)
@btime restrict($imgcolax, (1,2))
# before: 965.077 ns (18 allocations: 1.62 KiB)
# after:  690.381 ns (12 allocations: 1.33 KiB)
```

Reply:
Yeah, it's funny looking at old code I wrote! :laughing: Thanks for cleaning this up.